Update SRG-OS-000047-GPOS-00023 for RHEL 9 STIG

jan-cerny opened the pull request:
#### Description:
Update OCIL and add texts

#### Rationale:
RHEL 9 STIG

jan-cerny commented on linux_os/guide/system/auditing/configure_auditd_data_retention/auditd_data_retention_max_log_file_action/rule.yml:
```diff
@@ -55,12 +55,18 @@ references:
     maximum size:
     <tt>$ sudo grep max_log_file_action /etc/audit/auditd.conf</tt>
-    <pre>max_log_file_action <tt>rotate</tt></pre>
+    <pre>max_log_file_action = {{{ xccdf_value("var_auditd_max_log_file_action") }}}</pre>
```
Would other values be allowed, similar to other rules in this PR?

ggbecker replied:
the Oracle Linux 8 STIG has the following:

```
Details
Check Text (C-52161r779745_chk)

Verify that the SA and ISSO (at a minimum) are notified when the audit storage volume is full. 
 
Check which action OL 8 takes when the audit storage volume is full with the following command: 
 
$ sudo grep max_log_file_action /etc/audit/auditd.conf 
 
max_log_file_action=syslog 
 
If the value of the "max_log_file_action" option is set to "ignore", "rotate", or "suspend", or the line is commented out, this is a finding.
Fix Text (F-52115r779746_fix)

Configure OL 8 to notify the SA and ISSO when the audit storage volume is full by configuring the "max_log_file_action" parameter in the "/etc/audit/auditd.conf" file with the a value of "syslog" or "keep_logs": 
 
max_log_file_action=syslog
```

it prohibits the usage of "ignore", "rotate", or "suspend"

and allows  "syslog" or "keep_logs"

maybe we can use it as inspiration

jan-cerny replied:
I have add this  text here.

So now we have a different problem, in all rules in this PR:

The OCIL and fixtext isn't aligned with OVAL and Bash, because the OCIL and fixtext allow multiple values, but OVAL and bash uses specifically only the value set as variable in the SRG/control file.

Mab879 commented on controls/srg_gpos.yml:
```diff
@@ -35,8 +35,6 @@ controls:
             - var_auditd_space_left_percentage=25pc
             - var_auditd_space_left_action=email
-            - var_auditd_disk_error_action=halt
             - var_auditd_max_log_file_action=syslog
```
```suggestion

```

Mab879 commented on linux_os/guide/system/auditing/configure_auditd_data_retention/auditd_data_disk_full_action/rule.yml:
```diff
@@ -48,11 +48,24 @@ references:
+
+    disk_full_action = {{{ xccdf_value("var_auditd_disk_full_action") }}}
+
+    If availability has been determined to be more important, and this decision is documented with the ISSO, configure the operating system to notify system administration staff and ISSO staff in the event of an audit processing failure by setting the "disk_full_action" to "SYSLOG".
```
```suggestion
    If availability has been determined to be more important, and this decision is documented with the ISSO, configure {{{ full_name }}} to notify system administration staff and ISSO staff in the event of an audit processing failure by setting the "disk_full_action" to "SYSLOG".
```

Mab879 commented on linux_os/guide/system/auditing/configure_auditd_data_retention/auditd_data_disk_error_action/rule.yml:
```diff
@@ -40,11 +40,24 @@ references:
+
+    disk_error_action = {{{ xccdf_value("var_auditd_disk_error_action") }}}
+
+    If availability has been determined to be more important, and this decision is documented with the ISSO, configure the operating system to notify system administration staff and ISSO staff in the event of an audit processing failure by setting the "disk_error_action" to "SYSLOG".
```
```suggestion
    If availability has been determined to be more important, and this decision is documented with the ISSO, configure {{{ full_name }}} to notify system administration staff and ISSO staff in the event of an audit processing failure by setting the "disk_error_action" to "SYSLOG".
```

jan-cerny commented:
I have removed the variable and replaced the operating system by the full name.

But, the problem remains unsolved. Any suggestions how to solve this?

jan-cerny commented:
/retest

jan-cerny commented:
I have created STIG variants of the rules used by SRG-OS-000047-GPOS-00023 and I have rebased this PR on the top of the current upstream master branch.

ggbecker reviewed (dismissed):
We need to introduce the `srg_requirement` field for some of the rules added here:

`auditd_data_disk_full_action_stig`:  `The {{{ full_name }}} audit system must take appropriate action when the audit storage volume is full.`

Please check if other rules also require the field.

jan-cerny commented:
I have rebased this PR on the top of the latest upstream master and I have add srg_requirement to the rules.

jan-cerny commented:
I have resolved the conflicts and rebased this on the top of the latest upstream master branch.

jan-cerny commented:
Ihave add a missing new line at the end of the file
